### PR TITLE
Implement StatisticsItemEmptyView for AdManagement

### PR DESCRIPTION
**Pull request by robinsalehjan**
# Why?

- We were missing a `EmptyView` for the statistics section in the `AdManagement` screen.

# What?

- Implement the `EmptyView`

# Show me
|     Before              |            After               |
| -------------------- | ----------------------- |
| before | after |

### CODE REVIEW

File: Sources/Recycling/ListViews/AdManagement/Cells/UserAdManagementStatisticsCell.swift [hunk 1]
```diff
@@ -3,15 +3,28 @@
...
+    // MARK: - Public
+
     public var itemModels = [StatisticsItemModel]() {
         didSet { updateStackViewContent() }
     }
+
+    // MARK: - Private
+
+    private lazy var emptyView: StatisticsItemEmptyView = {
+        let view = StatisticsItemEmptyView(title: "Følg med på effekten", description: "Etter at du har publisert annonsen din kan du se statistikk for hvor mange som har sett annonsen din, favorisert den og som har fått tips om den.")
```
Comment (pimms) [inline Sources/Recycling/ListViews/AdManagement/Cells/UserAdManagementStatisticsCell.swift]: Should we have inline texts like this? I don't think it necessarily makes sense to inject "static" texts like these from the main app, but I also think it doesn't look right to have them hard-coded inline.

Would it make sense to have a `Localizable.strings` inside FinniversKit?

File: Sources/Components/StatisticsView/StatisticsItemEmptyView.swift [hunk 2]
```diff
@@ -0,0 +1,73 @@
...
+        let view = UIImageView(image: UIImage(named: .statsEmpty))
+        view.translatesAutoresizingMaskIntoConstraints = false
+        view.contentMode = .scaleAspectFit
+        view.tintColor = .ice
+        return view
+    }()
+
+    private lazy var titleLabel: UILabel = {
+        let label = UILabel()
+        label.translatesAutoresizingMaskIntoConstraints = false
```
Comment (bstien) [inline Sources/Components/StatisticsView/StatisticsItemEmptyView.swift]: You could use `UILabel(withAutoLayout: true)` instead

File: Sources/Components/StatisticsView/StatisticsItemEmptyView.swift [hunk 3]
```diff
@@ -0,0 +1,73 @@
...
+        label.font = UIFont.bodyStrong
+        label.textColor = .licorice
+        label.textAlignment = .center
+        label.setContentHuggingPriority(.defaultHigh, for: .vertical)
+        return label
+    }()
+
+    private lazy var descriptionLabel: UILabel = {
+        let label = UILabel()
+        label.translatesAutoresizingMaskIntoConstraints = false
```
Comment (bstien) [inline Sources/Components/StatisticsView/StatisticsItemEmptyView.swift]: Same here

File: Sources/Recycling/ListViews/AdManagement/Cells/UserAdManagementStatisticsCell.swift (hunk 1 above)
Comment (robinsalehjan) [reply Sources/Recycling/ListViews/AdManagement/Cells/UserAdManagementStatisticsCell.swift]: This isn't correct, brain fart.

The text is there for demo purposes, but in the actual app they will be in the `Localizable.strings` file.

I'll remove them from the constructor and make them part of the model or a separate model.

File: Sources/Recycling/ListViews/AdManagement/Cells/UserAdManagementStatisticsCell.swift [hunk 4]
```diff
@@ -33,17 +46,22 @@ public class UserAdManagementStatisticsCell: UITableViewCell {
...
+
+            for (index, model) in itemModels.enumerated() {
+                let itemView = StatisticsItemView(model: model)
+                itemView.shouldShowLeftSeparator = index > 0 && index < lastIndex
+                itemView.shouldShowRightSeparator = index > 0 && index < lastIndex
+                // Slight overengineering, as there are currently no plans to receive any other number than three items
+                if lastIndex == 1 { itemView.shouldShowLeftSeparator = index == lastIndex }
+                if lastIndex > 2 { itemView.shouldShowRightSeparator = index == lastIndex-1 }
+                stackView.addArrangedSubview(itemView)
+                itemView.setupConstraints()
```
Comment (bstien) [inline Sources/Recycling/ListViews/AdManagement/Cells/UserAdManagementStatisticsCell.swift]: Could you move this call to one of the lifecycle methods on the view itself? I.e. `viewDidLoad()`?

Comment (bstien) [reply Sources/Recycling/ListViews/AdManagement/Cells/UserAdManagementStatisticsCell.swift]: I see now that you didn't add this loc and that it's added to a stackView. Don't mind me :)

File: Demo/Recycling/ListViews/AdManagement/AdManagementDemoView.swift [hunk 5]
```diff
@@ -59,32 +65,40 @@ public class AdManagementDemoView: UIView {
...
 
     public func numberOfSections(in tableView: UITableView) -> Int {
-        return 1 + actionCellModels.count
+        return actionCellModels.count + 1
     }
 
     public func tableView(_ tableView: UITableView, cellForRowAt indexPath: IndexPath) -> UITableViewCell {
         if indexPath.section == 0 {
-            if indexPath.row == 0 {
+            if indexPath.row == 0 && statisticsCellModels.isEmpty == true {
```
Comment (nickfalk) [inline Demo/Recycling/ListViews/AdManagement/AdManagementDemoView.swift]: I think this if/else is uncalled for. :-)

**Pull request merged**